control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -389,6 +389,19 @@ func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv
 			}
 			return err
 		}
+
+		if aws.StringValue(output.VpcEndpoints[0].ServiceName) != awsEndpointService.Status.EndpointServiceName {
+			log.Info("conflicting endpoint found", "VPCEndpointServiceName", aws.StringValue(output.VpcEndpoints[0].ServiceName), "AWSEndpointServiceName", awsEndpointService.Status.EndpointServiceName)
```

Comment:
what's a conflicting endpoint why deletion is the action taken? can you elaborate?